src/utils/storage.ts

```diff
@@ -0,0 +1,83 @@
+import { ImporterState } from '../importer/types';
+import { SheetDefinition } from '../sheet/types';
+
+const DB_NAME = 'HelloCSV';
+const DB_VERSION = 1;
+const STORE_NAME = 'state';
+
+export async function getIndexedDBState(
+  sheetDefinitions: SheetDefinition[]
+): Promise<ImporterState> {
+  return new Promise((resolve, reject) => {
+    const key = getStateKey(sheetDefinitions);
+    const request = indexedDB.open(DB_NAME, DB_VERSION);
+
+    request.onerror = () => reject(request.error);
+    request.onsuccess = () => {
+      const db = request.result;
+      const transaction = db.transaction(STORE_NAME, 'readonly');
+      const store = transaction.objectStore(STORE_NAME);
+      const getRequest = store.get(key);
+
+      getRequest.onerror = () => reject(getRequest.error);
+      getRequest.onsuccess = () => {
+        try {
+          const result = getRequest.result;
+          result.sheetDefinitions = sheetDefinitions;
+          resolve(result);
+        } catch (error) {
+          reject(error);
+        }
+      };
+    };
+
+    request.onupgradeneeded = (event) => {
+      const db = (event.target as IDBOpenDBRequest).result;
+      if (!db.objectStoreNames.contains(STORE_NAME)) {
+        db.createObjectStore(STORE_NAME);
+      }
+    };
+  });
+}
+
+export async function setIndexedDBState(state: ImporterState): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const key = getStateKey(state.sheetDefinitions);
+    const value = { ...state } as any;
+    delete value.sheetDefinitions;
```

Comment:
sheetDefinitions have functions within, this should not be saved in the database